core/src/main/java/jenkins/mvn/FilePathGlobalSettingsProvider.java

```diff
@@ -32,34 +37,45 @@ public String getPath() {
     }
 
     @Override
-    public FilePath supplySettings(AbstractBuild<?, ?> build, TaskListener listener) {
+    public FilePath supplySettings(Run<?, ?> run, FilePath workspace, TaskListener listener) throws IOException, InterruptedException {
         if (StringUtils.isEmpty(path)) {
             return null;
         }
 
         try {
-            EnvVars env = build.getEnvironment(listener);
-            String targetPath = Util.replaceMacro(this.path, build.getBuildVariableResolver());
-            targetPath = env.expand(targetPath);
+            EnvVars env = run.getEnvironment(listener);
```

Comment:
Note that for Pipeline this will be the build’s global environment—it will not take into consideration `node`, `withEnv`, etc. If you actually want to use a local environment, you must pass in an `EnvVars`.